config/mail.php

```diff
@@ -38,6 +38,7 @@
     'mailers' => [
 
         'smtp' => [
+            'scheme' => env('MAIL_SCHEME', 'smtp'),
```

Comment:
This will make `scheme` default to `smtp` instead of relying on `port` value as indicated in #53585

With this changes everyone using the following configuration:

```ini
MAIL_ENCRYPTION=tls
MAIL_PORT=465
```

Will resolved to `stmp` instead correctly using `smtps` and create more gotcha than what we correctly have with `MAIL_ENCRYPTION`.

Reply:
Right now everyone who has not configured ```MAIL_ENCRYPTION=null``` has a broken configuration.

I also do not understand why we would be relying on a port to define a scheme that makes no sense.

Just because a port can be smtp or smtps doesn't mean it needs to be, 
I can have port 25 be smpts or I can have port 587 be smtp this is totally up to the setup of the provider.

Yes the defaults would be smtp  for 25 and smtps for 587 not to mention that 465 is a legacy default port for smtps nowadays it would/should be 587. 

So right now this would still end up causing issues for most people and having a dedicated configuration parameter would resolve this.

You can setup a fresh laravel installation and it will fail due to not having the proper config entries.

```env
MAIL_MAILER=smtp
MAIL_HOST=127.0.0.1
MAIL_PORT=9925
MAIL_USERNAME=null
MAIL_PASSWORD=null
MAIL_ENCRYPTION=null
MAIL_FROM_ADDRESS="hello@example.com"
MAIL_FROM_NAME="${APP_NAME}"
``` 
TestMail is a empty view Mailabable

```php
Mail::to('test@localhost')->send(new TestMail());
```
will fail with 
`The "" scheme is not supported; supported schemes for mailer "smtp" are: "smtp", "smtps".`

Reply:
> I also do not understand why we would be relying on a port to define a scheme that makes no sense.

This is a requirement by Symfony Mailer. `scheme` here is only used to comply with `Dsn` instance but the result is then passed to `EsmtpTransport` differently. 

https://github.com/symfony/mailer/blob/e4d358702fb66e4c8a2af08e90e7271a62de39cc/Transport/Smtp/EsmtpTransport.php#L56-L68

> The "" scheme is not supported; supported schemes for mailer "smtp" are: "smtp", "smtps".

The fixed in https://github.com/laravel/framework/pull/53585 has already fixed this and before the release can be done you should downgrade to 7.1 as suggested.

We will look into removing `MAIL_ENCRYPTION` and streamlining the configuration for next major version if possible.

Reply:
while #53585 might fix this for everyone using port 465 , this doesn't make much sense and it's still a breaking change because everyone with a port != 465 will have not ```smtps``` set on their configuration files while actually having ```MAIL_ENCRYPTION=tls```

Reply:
https://github.com/symfony/mailer/blob/7.2/CHANGELOG.md#440

`encryption` configuration was deprecated and removed in earlier `symfony/mailer` release. This probably being left out from `swiftmailer` era